asapdiscovery-data/asapdiscovery/data/selectors/pairwise_selector.py

```diff
@@ -0,0 +1,32 @@
+from itertools import product
+from typing import Literal, Union
+
+from asapdiscovery.data.schema_v2.complex import Complex, ComplexBase, PreppedComplex
+from asapdiscovery.data.schema_v2.ligand import Ligand
+from asapdiscovery.data.schema_v2.pairs import CompoundStructurePair, DockingInputPair
+from asapdiscovery.data.selectors.selector import SelectorBase
+
+
+class PairwiseSelector(SelectorBase):
+    """
+    Selects ligand and complex pairs by enumerating all possible pairs.
+    """
+
+    expander_type: Literal["PairwiseSelector"] = "PairwiseSelector"
+
+    def _select(
+        self, ligands: list[Ligand], complexes: list[Union[Complex, PreppedComplex]]
+    ) -> list[Union[CompoundStructurePair, DockingInputPair]]:
+        if not all(isinstance(c, ComplexBase) for c in complexes):
+            raise ValueError("All complexes must be of the same type")
+
+        pair_cls = self._pair_type_from_complex(complexes[0])
```

Comment:
maybe add some checking here that all `complexes` are the same class? might help prevent some weird errors down the line